File: source_funcs/go_transcoder/PROGRAM_FIND_REMAINDER_LARGE_NUMBER_DIVIDED_11/PROGRAM_FIND_REMAINDER_LARGE_NUMBER_DIVIDED_11.go
package main

func f_gold(str []byte) int {
	var (
		len_ int = int(len(str))
		num  int
		rem  int = 0
	)
	for i := int(0); i < len_; i++ {
		num = rem*10 + int(str[i]-byte('0'))
		rem = num % 11
	}
	return rem
}

File: rust_funcs/cpp_transcoder/PROGRAM_FIND_REMAINDER_LARGE_NUMBER_DIVIDED_11/PROGRAM_FIND_REMAINDER_LARGE_NUMBER_DIVIDED_11.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::str;

fn f_gold(str: String) -> i32  {
    let mut len = str.len() as i32;
    let mut num = 0;
    let mut rem = 0;
    for i in 0..min(len, 11) {
        num = rem * 10 + (str.chars().nth(i as usize).unwrap() as i32 - '0' as i32);
        rem = num % 11;
    }
    rem
}
